index.js

```diff
@@ -109,6 +109,15 @@ function getModuleDependencies (sources, module, queueName) {
     retval[match[2]].push(match[4])
   }
 
+  // force module to be integer, this can be important after uglify-js converted 1000 to 1e3
+  // in such case need to convert 1e3 back to 1000
+  var keys = Object.keys(retval);
+  for (var i = 0; i < keys.length; i++) {
+    for (var j = 0; j < retval[keys[i]].length; j++) {
+      retval[keys[i]][j] = +retval[keys[i]][j];
```

Comment:
Hi @ZuchaoWang 

Module id can be either integer or string so forcing it to be integer will break the library when the id is a string => `isNaN(+"abc") === true`. Unfortunately, the proposed solution won't work.

I guess that `retval[keys[i]][j] = retval[keys[i]][j] + '';` might work instead.

Or maybe doing the conversion after a check that `retval[keys[i]][j]` is a numeric value:
```js
function isNumeric(n) {
  return !isNaN(parseFloat(n)) && isFinite(n);
}
```

I'm not aware of edge cases in the proposed implementations so feel free to implement in a way you think will solve the issue in the best way.

Thank you for your contribution :)

Reply:
Maybe I should use the `all` option to handle this edge case. But I think there's a bug about `all`. In the following line of code, maybe ` { main: Object.keys(sources) }` should be replaced by `sources`.

`var requiredModules = options.all ? { main: Object.keys(sources) } : getRequiredModules(sources, moduleId)`

Otherwise `requiredModules` will become `{ main: { main: {...} } }`.

Reply:
In the documentation of webpack 4.10.2, it says module id is a number:

`Module ID is a number in webpack (in contrast to NodeJS where it is a string -- the filename).`

https://webpack.js.org/api/module-methods

Reply:
Hi @ZuchaoWang 

Module ID can be a string with the `NamedModulesPlugin` plugin. I think its enabled by default in development mode in Webpack 4.
In addition, if `DllPlugin` and `DllReferencePlugin` plugins used (which is supported by this library), the module ID becomes string which points to the correct DLL bundle.

You're right. There is a bug about `all` option. It should be `{ main: Object.keys(sources.main) }`. I'll fix it.

I didn't notice that the module ID extracted as a string by the RegExp regardless of its initial value so you're right and both my proposed implementations won't work.

What do you think of this solution:
```js
function isNumeric(n) {
  return !isNaN(1 * n); // 1 * n converts integers and 1e3 to integers and strings to NaN
}
...
  var keys = Object.keys(retval);
  for (var i = 0; i < keys.length; i++) {
    for (var j = 0; j < retval[keys[i]].length; j++) {
      if (isNumeric(retval[keys[i]][j])) {
        retval[keys[i]][j] =  1 * retval[keys[i]][j];
      }
    }
  }
```
?

Reply:
Hi @borisirota 

Your solution is almost Ok, except for the rare case where `NamedModulesPlugin` or `DllReferencePlugin` assigns a module id as string `1e3`. But I think this case can be ignored.

Reply:
@ZuchaoWang it won't happen since `NamedModulesPlugin` is always a path (relative paths starts with `./`) and `DllReferencePlugin` starts with "dll-reference" string so we are good to go :)

Do you want to update the PR so I can merge it?

Thanks!

Reply:
@borisirota That's good! I have updated the PR by adding your `isNumeric` solution.
For the `options.all` bug, I have kept it untouched.